src/vs/vscode.proposed.d.ts

```diff
@@ -2276,20 +2276,58 @@ declare module 'vscode' {
 
 	//#region https://github.com/Microsoft/vscode/issues/15178
 
-	// TODO@API must be a class
-	// TODO@API @lramos15 Call this XYZTabs
-	export interface OpenEditorInfo {
-		//viewColumn: ViewColumn; // todo @lramos15
-		name: string;
-		resource: Uri; // make optional @lramos15
+	/**
+	 * Represents a tab within the window
+	 */
+	export interface Tab {
+		/**
+		 * The text displayed on the tab
+		 */
+		label: string;
+
+		/**
+		 * The position of the tab
+		 */
+		viewColumn: ViewColumn;
+
+		/**
+		 * The resource represented by the tab.
+		 * May be undefined in certain cases such as diff editor and marketplace tabs
+		 */
+		resource?: Uri;
+
+		/**
+		 * Whether or not the tab is currently active
+		 * Dictated by being the selected tab in the active group
+		 */
 		isActive: boolean;
 	}
 
 	export namespace window {
-		export const openEditors: ReadonlyArray<OpenEditorInfo>;
+		/**
+		 * A list of all opened tabs
+		 * Ordered from left to right
+		 */
+		export const tabs: ReadonlyArray<Tab>;
```

Comment:
I wonder why we have `export let visibleTextEditors`, maybe @jrieken can comment on `let` vs `const`.

Reply:
🤷 actually not sure about that

Reply:
We now use `readonly Tab[]` over `ReadonlyArray` (unless things get too complex, which isn't the case here)